m4/ax_check_socket_options.m4

```diff
@@ -0,0 +1,186 @@
+dnl @synopsis Tests for socket options capability.
+dnl
+dnl Defines support for the following options:
+dnl     SO_KEEPALIVE
+dnl     TCP_KEEPALIVE
+dnl     TCP_KEEPCNT
+dnl     TCP_KEEPIDLE
+dnl     TCP_KEEPINTVL
+dnl
+dnl The macro is a modified version of the acinclude.m4 from the ZeroMQ library
+dnl http://github.com/zeromq/libzmq
+dnl @license Mozilla Public License v2 (MPLv2)
+
+
+dnl ################################################################################
+dnl # CHECK_SO_KEEPALIVE([action-if-found], [action-if-not-found])          #
```

Comment:
Documentation headers looks wrong. The `#` might be moved to the end of the line. If my comment applies then you should fix the other documentation headers too.